### PR TITLE
LibWeb: Implement table rowspan

### LINKED ISSUE #18952: LibWeb: Table cells don't follow the rowspan attribute
The following example shows the problem for cells which specify a `rowspan`:

https://mdn.github.io/learning-area/html/tables/assessment-finished/planets-data.html

The HTML specifies several `rowspan`s:

```
<th rowspan="4" colspan="2" scope="rowgroup">Terrestrial planets</th>
...
<th rowspan="4" scope="rowgroup">Jovian planets</th>
<th rowspan="2" scope="rowgroup">Gas giants</th>
...
<th rowspan="2" scope="rowgroup">Ice giants</th>
```

Ladybird (cropped to the relevant section):

ladybird

Expected (cropped to the relevant section):

chrome

### PR DESCRIPTION
**Pull request by axgallo**
Adjust computing the table height and positioning of cells to account for the rowspan property.

Fixes #18952.

### CODE REVIEW

**Comment by axgallo** [comment]
I'm not sure how to test this yet, since the content height of cells spanning multiple rows stays the same, only the top and bottom padding change.

File: Userland/Libraries/LibWeb/Layout/TableFormattingContext.cpp [hunk 1]
```diff
@@ -671,6 +675,28 @@ void TableFormattingContext::position_cell_boxes()
...
+        if (i == 0) {
+            span_height += row_state.content_height() + row_state.border_box_bottom();
+        } else if (i == cell.row_span - 1) {
+            span_height += row_state.border_box_top() + row_state.content_height();
+        } else {
+            span_height += row_state.border_box_height();
+        }
+    }
+    return span_height;
+}
```
Comment (kalenikaliaksandr) [inline Userland/Libraries/LibWeb/Layout/TableFormattingContext.cpp:698]: initially, it be difficult to understand why borders are not taken in account when then span is equal to 1, but they are included in content height when span > 1. some additional clarification would be helpful.

**Comment by kalenikaliaksandr** [comment]
very cool! this change seems to make a lot of tables on wikipedia look better :)

File: Userland/Libraries/LibWeb/Layout/TableFormattingContext.cpp (hunk 1 above)
Comment (axgallo) [reply Userland/Libraries/LibWeb/Layout/TableFormattingContext.cpp:698]: When the row span is greater than 1, the borders of inner rows within the span have to be included in the content height of the spanning cell. The first top and final bottom are excluded to be consistent with the row span of 1, which excludes the top and bottom border of the single cell (we use the content height in that case).

I'm not sure I got it right, does it make sense to you?

Comment (kalenikaliaksandr) [reply Userland/Libraries/LibWeb/Layout/TableFormattingContext.cpp:698]: yes, that makes sense. if you can put comment that would be nice :)

**Comment by axgallo** [comment]
> > I'm not sure how to add a layout test for this yet, since the content height of cells spanning multiple rows stays the same, only the top and bottom padding change.
> 
> It is possible to test this change with layout test. The content of cells has position that is present in layout dump. I created simple test for this change and that's how layout dumps diff of master and your branch looks:
> 
> ```diff
> @@ -39,9 +39,9 @@
>              TableRowBox <tr> at (8,27.46875) content-size 221.359375x19.46875 children: not-inline
>                BlockContainer <(anonymous)> at (8,8) content-size 0x0 children: inline
>                  TextNode <#text>
> -              TableCellBox <td> at (9,28.46875) content-size 70.046875x17.46875 [BFC] children: inline
> +              TableCellBox <td> at (9,38.203125) content-size 70.046875x17.46875 [BFC] children: inline
>                  line 0 width: 49.609375, height: 17.46875, bottom: 17.46875, baseline: 13.53125
> -                  frag 0 from TextNode start: 0, length: 5, rect: [9,28.46875 49.609375x17.46875]
> +                  frag 0 from TextNode start: 0, length: 5, rect: [9,38.203125 49.609375x17.46875]
>                      "Row 1"
>                  TextNode <#text>
>                BlockContainer <(anonymous)> at (8,8) content-size 0x0 children: inline
> ```

Could you share the test? I have an alternative way in a local branch which adds an option to the headless browser which dumps the box model too and adjusts `layout_test.sh` to enable it based on a magic first line comment in the input HTML, but that might be overkill if you have a way to test this without it.

**Comment by kalenikaliaksandr** [comment]
That is html I used for the test:
```html
<!DOCTYPE html>
<html>
    <head>
        <title>Table with rowspan</title>
    </head>
    <body>
        <table border="1">
            <tr>
                <th>Header 1</th>
                <th>Header 2</th>
                <th>Header 3</th>
            </tr>
            <tr>
                <td rowspan="2">Row 1</td>
                <td>Cell 1</td>
                <td>Cell 2</td>
            </tr>
            <tr>
                <td>Cell 3</td>
                <td>Cell 4</td>
            </tr>
            <tr>
                <td>Row 2</td>
                <td>Cell 5</td>
                <td>Cell 6</td>
            </tr>
        </table>
    </body>
</html>
```

File: Userland/Libraries/LibWeb/Layout/TableFormattingContext.cpp (hunk 1 above)
Comment (axgallo) [reply Userland/Libraries/LibWeb/Layout/TableFormattingContext.cpp]: Added a comment.

**Review by kalenikaliaksandr — APPROVED** [review]

**Pull request merged**